### PR TITLE
chore: Add comprehensive test cases for RequestValidator to improve coverage and debugging #902

### LINKED ISSUE #645: Add more RequestValidator test cases
### Issue Summary

RequestValidator only has one example in its test, [RequestValidatorTest](https://github.com/twilio/twilio-java/blob/0e4637bb772209737e3a689665b8a07cd99b3d58/src/test/java/com/twilio/security/RequestValidatorTest.java).

I'm attempting to use RequestValidator but it's not working - it says all requests that come from Twilio are invalid. As there's only one example that makes it very difficult to diagnose what the problem is. It would help massively to have more examples.

As RequestValidator is part of security, good test coverage is really important, especially edge cases.

Here's a starting list of the test cases to be added

* A request with and without form parameters
* A request with and without query parameters
* A request with and without non-Twilio query parameters (e.g. I set a Voice Callback URL to be `https://my-website.abc/twilio/callback/?type=thiswontbeusedinthesignaturegeneratorandsoshouldbeignored`
* A request with URL encoded query parameters

And just a lot more tests in general. At the moment the RequestValidator is unusable.

### Technical details:
* twilio-java version: 8.19.1
* java version: 11

### PR DESCRIPTION
**Pull request by manisha1997**
The RequestValidator test suite previously had only 11 basic test cases, making it difficult for developers to diagnose validation issues and understand expected behavior in edge cases. This PR significantly expands test coverage by adding 17 new comprehensive test cases.

## What's Changed

**New test coverage includes:**

- **Empty and null parameter handling** - Tests validation with empty maps, null parameters, and missing form data
- **Query parameter scenarios** - URLs with and without query parameters, including URL-encoded values and non-Twilio parameters
- **Edge case data handling** - Special characters, unicode, very long values (1000+ chars), empty/null parameter values
- **Behavioral verification** - Parameter key case sensitivity, alphabetical sorting, international characters in URLs
- **Error conditions** - Invalid, null, and empty signatures to ensure proper failure modes

**Example of improved debugging capabilities:**

```java
// Before: Only basic validation test
@Test
public void testValidate() {
    Assert.assertTrue("Request does not match provided signature", 
        validator.validate(url, params, signature));
}

// After: Comprehensive edge case coverage
@Test
public void testValidateWithSpecialCharactersInParams() {
    Map<String, String> specialParams = new HashMap<>();
    specialParams.put("Message", "Hello & goodbye! @#$%^*()");
    specialParams.put("Special", "unicode: ñáéíóú");
    
    String expectedSignature = "dCPiR4WtQ6QFN6pJh81CtlCcWLQ=";
    boolean isValid = validator.validate(url, specialParams, expectedSignature);
    
    Assert.assertTrue("Validation should succeed with special characters", isValid);
}
```

## Impact

- **Test count increased from 11 to 28** - More than doubled test coverage
- **Better developer experience** - Clear examples of expected behavior across various scenarios
- **Improved reliability** - Edge cases that could cause real-world validation failures are now tested
- **Enhanced security** - Comprehensive testing of the security-critical RequestValidator component

All new test cases use correctly calculated signatures and follow existing project conventions. The expanded test suite makes RequestValidator much more usable and debuggable for developers integrating Twilio webhooks.

Fixes #645.

---

💬 Share your feedback on Copilot coding agent for the chance to win a $200 gift card! Click [here](https://survey3.medallia.com/?EAHeSx-AP01bZqG0Ld9QLQ) to start the survey.

### CODE REVIEW

**Review by sbansla — APPROVED** [review]

File: src/test/java/com/twilio/security/RequestValidatorTest.java [hunk 1]
```diff
@@ -114,4 +114,180 @@ public void testValidateAddsPortHttp() {
...
+        boolean isValid = validator.validate(url, emptyParams, expectedSignature);
+
+        Assert.assertTrue("Validation should succeed with empty parameters", isValid);
+    }
+
+    @Test
+    public void testValidateWithNullParams() {
+        // Test with null parameters
+        String expectedSignature = "zYQTYrRWXE7LtzbG4PfP7/bkkGo="; // hash of URL only
+        boolean isValid = validator.validate(url, (Map<String, String>) null, expectedSignature);
```
Comment (shrutiburman) [inline src/test/java/com/twilio/security/RequestValidatorTest.java:132]: what if we dont type-cast null, we should expect a NPE?
